pkg/registry/file/vulnsummarystorage.go

```diff
@@ -0,0 +1,240 @@
+		return nil, err
+	}
+	err = json.Unmarshal(b, &vulnManifestSummary)
+	if err != nil {
+		logger.L().Ctx(ctx).Error("json unmarshal failed", helpers.Error(err), helpers.String("key", path))
+		return nil, err
+	}
+	if _, exist := vulnManifestSummary.GetLabels()["kubescape.io/workload-namespace"]; !exist {
+		return nil, storage.NewKeyNotFoundError(path, 0)
+	}
+	return &vulnManifestSummary, nil
+}
+
+// add uni test
+func initNamespaceVulnSummary(key string) softwarecomposition.VulnerabilitySummary {
+	namespace := filepath.Base(key)
+	var vulnSummaryByNamespace softwarecomposition.VulnerabilitySummary
+	vulnSummaryByNamespace.APIVersion = "spdx.softwarecomposition.kubescape.io/v1beta1"
+	vulnSummaryByNamespace.Kind = "VulnerabilitySummary"
+	vulnSummaryByNamespace.Labels = map[string]string{"kubescape.io/workload-namespace": namespace}
+	vulnSummaryByNamespace.Annotations = map[string]string{"kubescape.io/status": ""}
+	vulnSummaryByNamespace.Name = namespace
+	vulnSummaryByNamespace.Namespace = namespace
+	vulnSummaryByNamespace.CreationTimestamp.Time = time.Now()
+
+	return vulnSummaryByNamespace
+}
+
+func updateVulnCounters(aggregatedCounters *softwarecomposition.VulnerabilityCounters, counters *softwarecomposition.VulnerabilityCounters) {
+	aggregatedCounters.All += counters.All
+	aggregatedCounters.Relevant += counters.Relevant
+}
+
+func updateSeverities(aggregatedSeverities *softwarecomposition.SeveritySummary, severities *softwarecomposition.SeveritySummary) {
+	updateVulnCounters(&aggregatedSeverities.Critical, &severities.Critical)
+	updateVulnCounters(&aggregatedSeverities.High, &severities.High)
+	updateVulnCounters(&aggregatedSeverities.Medium, &severities.Medium)
+	updateVulnCounters(&aggregatedSeverities.Low, &severities.Low)
+	updateVulnCounters(&aggregatedSeverities.Negligible, &severities.Negligible)
+	updateVulnCounters(&aggregatedSeverities.Unknown, &severities.Unknown)
+}
+
+func updateFullVulnSumm(fullVulnSumm *softwarecomposition.VulnerabilitySummary, vulnManifestSumm *softwarecomposition.VulnerabilityManifestSummary) {
+	updateSeverities(&fullVulnSumm.Spec.Severities, &vulnManifestSumm.Spec.Severities)
+	if vulnManifestSumm.Spec.Vulnerabilities.ImageVulnerabilitiesObj != (softwarecomposition.VulnerabilitiesObjScope{}) {
+		fullVulnSumm.Spec.WorkloadVulnerabilitiesObj = append(fullVulnSumm.Spec.WorkloadVulnerabilitiesObj, vulnManifestSumm.Spec.Vulnerabilities.ImageVulnerabilitiesObj)
+	}
+	if vulnManifestSumm.Spec.Vulnerabilities.WorkloadVulnerabilitiesObj != (softwarecomposition.VulnerabilitiesObjScope{}) {
+		fullVulnSumm.Spec.WorkloadVulnerabilitiesObj = append(fullVulnSumm.Spec.WorkloadVulnerabilitiesObj, vulnManifestSumm.Spec.Vulnerabilities.WorkloadVulnerabilitiesObj)
+	}
+}
+
+func (s *VulnSummaryStorageImpl) getVulnManifestSummaryDirPath(key string) string {
+	return filepath.Join(s.root, key, "..", "..", "..", "vulnerabilitymanifestsummaries")
```

Comment:
issue: function jumps up 3 levels of “path” without being tested

In general, the “key” being passed is not a file path. It is a concidence that we use it to store the file within the file storage implementation. But this isn’t the file storage. This is the Vulnerability Manifest storage, and it should have no business dealing with files or any file-related logic.

Instead, it makes more sense to re-evaluate what we need to build a summary of vulnerabilities. To me, it seems that we need at least the name of the scope that we are building for (be it namespace or cluster) and a way to fetch vulnerability summaries by a filter like namespace. Then, this function would not be needed.